cmd/upload/upload.go

```diff
@@ -473,6 +478,10 @@ func (app *UpCmd) handleAsset(ctx context.Context, a *browser.LocalAssetFile) er
 			return nil
 		}
 		app.manageAssetAlbum(ctx, ID, a, advice)
+		if app.AutoArchive && a.Archived {
+			a.Archived = true
+			app.Immich.UpdateAsset(ctx, ID, a)
+		}
```

Comment:
I think you can simplify this code.

The AssetUpload method of the immich client must be updated to handle the Archived flag: 
https://github.com/simulot/immich-go/blob/d58d2669ae6f13bbb32ff45fc787e92f0ae4d166/immich/asset.go#L125-L127

Doing so, you'll save one call

Reply:
Also add commit that mention the fix of the #448